src/main/drivers/rangefinder/rangefinder_lidartf.c

```diff
@@ -0,0 +1,276 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "platform.h"
+
+#ifdef USE_RANGEFINDER_TF
+
+#include "build/debug.h"
+#include "build/build_config.h"
+
+#include "io/serial.h"
+
+#include "drivers/time.h"
+#include "drivers/rangefinder/rangefinder.h"
+#include "drivers/rangefinder/rangefinder_lidartf.h"
+
+#define TF_DEVTYPE_NONE 0
+#define TF_DEVTYPE_MINI 1
+#define TF_DEVTYPE_02   2
+
+static uint8_t tfDevtype = TF_DEVTYPE_NONE;
+
+#define TF_FRAME_LENGTH    6             // Exclusing sync bytes (0x59) x 2 and checksum
+#define TF_FRAME_SYNC_BYTE 0x59
+#define TF_TIMEOUT_MS      (100 * 2)
+
+//
+// Benewake TFmini frame format
+// Byte
+// -: SYNC
+// -: SYNC
+// 0: Measured distance (LSB)
+// 1: Measured distance (MSB)
+// 2: Signal strength (LSB)
+// 3: Signal strength (MSB)
+// 4: Integral time
+// 5: Reserved
+// -: Checksum (Unsigned 8-bit sum of bytes 0~7)
+//
+// Note *1: TFmini product specification (Version A00) specifies byte 6 is
+// reserved and byte 7 is quality, but it seems like byte 7 contains
+// value of 7 for good measurement and 2 for bad measurement.
+//
+// Credibility
+// 1. If distance is 12m (1200cm), then OoR.
+//
+#define TF_MINI_FRAME_INTEGRAL_TIME 4
+#define TF_MINI_RANGE_MIN 40
+#define TF_MINI_RANGE_MAX 1200
+#define TF_DETECTION_CONE_DECIDEGREES 900 // Very conservative.
```

Comment:
Does `conservative` mean that distance reading is reliable when quad is tilted by 45 deg?

Reply:
I think as long as credible reading is obtained, it can be tilted almost vertically. Isn't this what detection cone means?